packages/widgets/src/HeaderBar/Apps.js

```diff
@@ -84,194 +83,218 @@ Search.propTypes = {
+                    filter
+                ).toLowerCase()
+
+                return filter.length > 0
+                    ? formattedAppName.match(formattedFilter)
+                    : true
+            })
+            .map(({ displayName, name, defaultAction, icon }, idx) => (
+                <Item
+                    key={`app-${name}-${idx}`}
+                    name={displayName || name}
+                    path={defaultAction}
+                    img={icon}
+                />
+            ))}
+
+        <style jsx>{`
+            div {
+                display: flex;
+                flex-direction: row;
+                flex-wrap: wrap;
+                align-content: flex-start;
+                align-items: flex-start;
+                justify-content: flex-start;
+                width: 30vw;
+                min-width: 300px;
+                max-width: 560px;
+
+                min-height: 200px;
+                max-height: 465px;
+                margin: 0 8px 8px 8px;
+
+                overflow: auto;
+                overflow-x: hidden;
+            }
+        `}</style>
+    </div>
+)
 
-            <style jsx>{`
-                div {
-                    display: flex;
-                    flex-direction: row;
-                    flex-wrap: wrap;
-                    align-content: flex-start;
-                    align-items: flex-start;
-                    justify-content: flex-start;
-                    width: 30vw;
-                    min-width: 300px;
-                    max-width: 560px;
-
-                    min-height: 200px;
-                    max-height: 465px;
-                    margin: 0 8px 8px 8px;
-
-                    overflow: auto;
-                    overflow-x: hidden;
-                }
-            `}</style>
-        </div>
-    )
-}
 List.propTypes = {
     apps: propTypes.array,
     filter: propTypes.string,
 }
 
-export default class Apps extends React.Component {
-    state = {
-        show: false,
-        filter: '',
-    }
-
-    componentDidMount() {
-        document.addEventListener('click', this.onDocClick)
-    }
+const AppMenu = ({ apps, filter, onFilterChange }) => (
+    <div data-test="headerbar-apps-menu">
+        <Card>
+            <Search value={filter} onChange={onFilterChange} />
+            <List apps={apps} filter={filter} />
+        </Card>
+
+        <style jsx>{`
+            div {
+                z-index: 10000;
+                position: absolute;
+                top: 28px;
+                right: -6px;
+                border-top: 4px solid transparent;
+            }
+        `}</style>
+    </div>
+)
+
+AppMenu.propTypes = {
+    apps: propTypes.array.isRequired,
+    onFilterChange: propTypes.func.isRequired,
+    filter: propTypes.string,
+}
 
-    componentWillUnmount() {
-        document.removeEventListener('click', this.onDocClick)
-    }
+const mutation = {
+    resource: 'i18n',
+    type: 'create',
+    data: ({ items }) => items,
+}
```

Comment:
Hmmm... logically this is not a mutation, it's a query, the backend just uses a POST endpoint to support a long list of apps.  Ideally we should use a `Query` here and transparently translate it into a POST request in the data engine - the consumer shouldn't have to care about how the data is actually fetched.

I think it's OK to "break the glass" here and pretend it's a mutation, but we'll want to fix it properly soon so if we don't do that now let's create an issue to track it.